File: 2503/app/src/main/java/com/example/a2503/MainActivity.java
package com.example.a2503;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textview.MaterialTextView;

public class MainActivity extends AppCompatActivity {

    private TextInputEditText titleEditText, genreEditText, yearEditText, searchEditText;
    private MaterialButton addButton, searchButton;
    private MaterialTextView searchResultTextView;
    private DatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        titleEditText = findViewById(R.id.titleEditText);
        genreEditText = findViewById(R.id.genreEditText);
        yearEditText = findViewById(R.id.yearEditText);
        searchEditText = findViewById(R.id.searchEditText);
        addButton = findViewById(R.id.addButton);
        searchButton = findViewById(R.id.searchButton);
        searchResultTextView = findViewById(R.id.searchResultTextView);

        dbHelper = new DatabaseHelper(this);

        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String title = titleEditText.getText().toString().trim();
                String genre = genreEditText.getText().toString().trim();
                String yearStr = yearEditText.getText().toString().trim();

                if (title.isEmpty() || genre.isEmpty() || yearStr.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Заполните все поля", Toast.LENGTH_SHORT).show();
                    return;
                }

                int year;
                try {
                    year = Integer.parseInt(yearStr);
                } catch (NumberFormatException e) {
                    Toast.makeText(MainActivity.this, "Введите год корректно", Toast.LENGTH_SHORT).show();
                    return;
                }

                boolean success = dbHelper.addMovie(title, genre, year);
                if (success) {
                    Toast.makeText(MainActivity.this, "Фильм успешно добавлен", Toast.LENGTH_SHORT).show();
                    titleEditText.setText("");
                    genreEditText.setText("");
                    yearEditText.setText("");
                } else {
                    Toast.makeText(MainActivity.this, "Не удалось добавить фильм", Toast.LENGTH_SHORT).show();
                }
            }
        });

        searchButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String searchTitle = searchEditText.getText().toString().trim();
                if (searchTitle.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Введите имя фильма для поиска", Toast.LENGTH_SHORT).show();
                    return;
                }

                String result = dbHelper.searchMovie(searchTitle);
                searchResultTextView.setText(result);
            }
        });
    }
}

File: 04_06/app/src/main/java/com/example/a04_06/MainActivity.kt
package com.example.a04_06

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {

    private var isOriginalOrder = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container_1, FirstFragment())
            .replace(R.id.fragment_container_2, SecondFragment())
            .commit()

        val swapButton = findViewById<Button>(R.id.swapButton)
        swapButton.setOnClickListener {
            swapFragments()
        }
    }

    private fun swapFragments() {
        val first: Fragment = if (isOriginalOrder) SecondFragment() else FirstFragment()
        val second: Fragment = if (isOriginalOrder) FirstFragment() else SecondFragment()

        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container_1, first)
            .replace(R.id.fragment_container_2, second)
            .commit()

        isOriginalOrder = !isOriginalOrder
    }
}
